app/controllers/agent/ManageAgentController.scala

```diff
@@ -142,59 +158,128 @@ class ManageAgentController @Inject()(
-          Some(
-            unassignAgentFromPropertyView(
-              submitAgentAppointmentRequest,
-              agent,
-              calculateBackLink(organisationsAgents, agent.representativeCode)))
+      agentToBeManagedOpt match {
+        case None => throw new NotFoundException("Agent is not available")
+        case Some(agent) =>
+          if (featureSwitch.isAgentListYearsEnabled) {
+            val ManageAgentOptionItemList =
+              (ipPropertyLinksCount, agentToBeManagedOpt) match {
+                case (0, Some(agent)) if agent.propertyCount == 0 =>
+                  List(
+                    ManageAgentOptionItem(ChangeRatingList),
+                    ManageAgentOptionItem(RemoveFromYourAccount)
+                  )
+                case (1, Some(agent)) if agent.propertyCount == 0 =>
+                  List(
+                    ManageAgentOptionItem(AssignToYourProperty),
+                    ManageAgentOptionItem(ChangeRatingList),
+                    ManageAgentOptionItem(RemoveFromYourAccount)
+                  )
+                case (1, Some(agent)) if agent.propertyCount == 1 =>
+                  List(
+                    ManageAgentOptionItem(UnassignFromYourProperty),
+                    ManageAgentOptionItem(ChangeRatingList)
+                  )
+                case (numberOfPropertyLinks, Some(agent)) if numberOfPropertyLinks > 1 && agent.propertyCount == 0 =>
+                  List(
+                    ManageAgentOptionItem(AssignToAllProperties),
+                    ManageAgentOptionItem(AssignToSomeProperties),
+                    ManageAgentOptionItem(ChangeRatingList),
+                    ManageAgentOptionItem(RemoveFromYourAccount),
+                  )
+                case (numberOfPropertyLinks, Some(agent)) if numberOfPropertyLinks > agent.propertyCount =>
+                  List(
+                    ManageAgentOptionItem(AssignToAllProperties),
+                    ManageAgentOptionItem(AssignToSomeProperties),
+                    ManageAgentOptionItem(UnassignFromAllProperties),
+                    ManageAgentOptionItem(UnassignFromSomeProperties),
+                    ManageAgentOptionItem(ChangeRatingList)
+                  )
+                case (numberOfPropertyLinks, Some(agent)) if numberOfPropertyLinks == agent.propertyCount =>
+                  List(
+                    ManageAgentOptionItem(UnassignFromAllProperties),
+                    ManageAgentOptionItem(UnassignFromSomeProperties),
+                    ManageAgentOptionItem(ChangeRatingList)
+                  )
+              }
+            Some(
+              manageAgentView(
+                submitManageAgentForm,
+                ManageAgentOptionItemList,
+                agent,
+                calculateBackLink(organisationsAgents, agent.representativeCode)
+              ))
+          } else {
+            (ipPropertyLinksCount, agentToBeManagedOpt) match {
```

Comment:
`agent` is already available here taken out of the option so you don't need to match on it again.